sdk/storage/inc/shares/protocol/share_rest_client.hpp

```diff
@@ -0,0 +1,6898 @@
+    std::string FileCreationTime;
+    std::string FileLastWriteTime;
+    std::string FileChangeTime;
+    std::string FilePermissionKey;
+    std::string FileId;
+    std::string FileParentId;
+    LeaseDurationType LeaseDuration;
+    LeaseStateType LeaseState;
+    LeaseStatusType LeaseStatus;
+  };
+
+  struct FileDeleteResponse
+  {
+  };
+
+  struct FileSetHTTPHeadersResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    bool IsServerEncrypted = bool();
+    std::string FilePermissionKey;
+    std::string FileAttributes;
+    std::string FileCreationTime;
+    std::string FileLastWriteTime;
+    std::string FileChangeTime;
+    std::string FileId;
+    std::string FileParentId;
+  };
+
+  struct FileSetMetadataResponse
+  {
+    std::string ETag;
+    bool IsServerEncrypted = bool();
+  };
+
+  struct FileAcquireLeaseResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string LeaseId;
+  };
+
+  struct FileReleaseLeaseResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+  };
+
+  struct FileChangeLeaseResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string LeaseId;
+  };
+
+  struct FileBreakLeaseResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string LeaseId;
+  };
+
+  struct FileUploadRangeResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string ContentMD5;
+    bool IsServerEncrypted = bool();
+  };
+
+  struct FileUploadRangeFromURLResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string XMsContentCrc64;
+    bool IsServerEncrypted = bool();
+  };
+
+  struct FileGetRangeListResponse
+  {
+    std::vector<Range> RangeList;
+    std::string LastModified;
+    std::string ETag;
+    int64_t FileContentLength = int64_t();
+  };
+
+  struct FileStartCopyResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string CopyId;
+    CopyStatusType CopyStatus;
+  };
+
+  struct FileAbortCopyResponse
+  {
+  };
+
+  struct FileListHandlesResponse
+  {
+    std::vector<HandleItem> HandleList;
+    std::string NextMarker;
+    ShareFileHttpHeaders HttpHeaders;
+  };
+
+  struct FileForceCloseHandlesResponse
+  {
+    std::string marker;
+    int32_t numberOfHandlesClosed = int32_t();
+    int32_t numberOfHandlesFailedToClose = int32_t();
+  };
+
+  class ShareRestClient {
+  public:
+    class Service {
+    public:
+      struct SetPropertiesOptions
+      {
+        StorageServiceProperties ServiceProperties; // The StorageService properties.
+        Azure::Core::Nullable<int32_t>
+            Timeout; // The timeout parameter is expressed in seconds. For more information, see <a
+                     // href="https://docs.microsoft.com/en-us/rest/api/storageservices/Setting-Timeouts-for-File-Service-Operations?redirectedfrom=MSDN">Setting
+                     // Timeouts for File Service Operations.</a>
+        std::string ApiVersionParameter
```

Comment:
why is this in options? Do we allow users to specify API version?

Reply:
It's never exposed to convenience layer.

Reply:
I don't think it's ok. If our customers saw this, they'd think we're planning on support this feature, which we're not.

Reply:
Added this into the discussion topic list.